### DISCUSSION TITLE
First & bad use of git LFS - can't push anymore

### DISCUSSION THREAD

**Discussion by Yasgal** [git-lfs/git-lfs#5918] [Q&A]
Hello everyone,

I'm really in distress, I just blocked my whole group on a game production, because I installed GIT LFS due to my lightning data making 103mo (baking).

I told him to take only two types of file in the scenes folder, .asset for the lightning data and .png for the lightning maps.

However, when I made my push I got an error saying that my LFS GIT was FULL.

Trying to look, I don't know how but it took a lot of stuff from other branches including mine and with the 2 file types tracked.

I tried to uninstall GIT LFS to start again at 0 but I could not because I have a “pre push” and to remove it I have to find where it is... But I don't know what it's talking about and I can't find anything on the internet.

2 people in my group tried to pull, and they got an error with GIT LFS because I tried to delete with the “plum” command. From what I understand, the error is due to the fact that the GIT LFS is full, and maybe some files that I deleted are missing, and we can't do anything, neither pull nor push.

I also did the “uninstall git lfs” command in the 2 places where I had done it, it tells me that it's deleted but at the same time I get an error.

On Fork to succeed in pushing I have to “Deinitialized git lfs” and it works but it comes back as soon as the next push.

I really screwed up, I didn't think it could be so hard to set something up just to have a little more room...

No one in my class has ever used git lfs and I'm all alone with this error on my shoulders.

How can I just erase all without losing files on my project and make a good set up ? Because for my lightning data I need git LFS for sure.

**Comment by chrisd8088** [comment]
Hey, I'm sorry you're having trouble.

You don't mention which Git hosting service you're using (perhaps GitHub or GitLab?), but you could always try reaching out to their support team for assistance as well.  This project is for the open-source Git LFS client, so we can't always provide advice specific to a given hosting service.

That said, I can try to provide some general advice here, based on what you've described.  It's always helpful if you can provide the exact commands you're using, the specific error messages they output, and the output of running `git lfs env` so that we know more about your system configuration.

You mention trying to restrict Git LFS to tracking some specific types of files.  That would normally be done with a command sequence like the following, but note that this will apply to all `*.asset` and `*.png` files in the folder.  If you need more specific tracking, you can just use a single file name instead of a wildcard `*` character, for example.
```
$ git lfs track '/path/to/scenes/folder/*.asset'
$ git lfs track '/path/to/scenes/folder/*.png'
$ git add .gitattributes
$ git commit -m 'add gitattributes for Git LFS'
```

Now, to be clear, that will mean any files which match those filename patterns and which are added in subsequent commits will be converted to Git LFS objects.  However, if you already have files matching those patterns in previous commits in your Git history, those won't be automatically converted into Git LFS objects; for that you'd need to use `git lfs migrate import` to rewrite your Git history.  Given that you're having some trouble now, though, I would not recommend that step.  It's important to understand how Git and Git LFS work before rewriting your Git history.

If you've tried to uninstall Git LFS, and the pre-push hook is still in place, you can find the location where Git hooks are installed using `git rev-parse --git-path hooks`.  Then delete the `pre-push` file in that location, or (if you want to be careful) move the file or rename it so it's not called `pre-push` anymore.

For reference, the Git manual has a [page](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) describing how hook scripts work in general.  Git LFS installs a pre-push hook script because that's how it makes sure it can push all the Git LFS objects into the remote's storage before the actual Git push occurs.

This "How Git LFS Works" [blog post](https://jaredkhan.com/blog/how-git-lfs-works) is a really nice description of the whole process!

You mention the `plum` command, but I don't know what that might be.  Perhaps you meant `git lfs prune`?

As for resetting things to a default state, that does depend on exactly what your current situation is.

If you've run `git lfs uninstall` and then made sure the `pre-push` hook script has been deleted (or moved or renamed), that's one step, of course.  It will mean that you're not able to run a command like `git lfs ls-files`, which might tell you which files you've converted into Git LFS objects, so you may want to hold off a bit until you've got an understanding of what's happened in your Git history.

That's maybe the key point I'd stress: a Git repository has a history, which is available with standard Git commands like `git log`.  If you still have Git LFS installed, `git lfs ls-files` is useful as well to see which files are currently being tracked.  Using commands like those to get a detailed picture of where the Git LFS objects are in your file hierarchy, and in which commits and branches they appear, will be most useful in unwinding any changes you've made that you want to undo.

I hope those notes are of some assistance.  The more you can share regarding the exact state of your repository, the more we should be able to help.  And please do get in touch with the support team for the Git hosting service you're using as they may be able to give more help too, by looking at your repository (if you give them permission to do so).

**Reply by Yasgal** [reply]
We managed to create another repository ahah, I've a lot of command without fixing my issues, but I was more easy to start in a cleaner way without LFS.

I'm just wondering how I would managed my lightning data high quality at the end of the production, maybe by doing it and build the game without pushing.

Thank you a lot for your help, I learn more about git LFS, but at the same time this make me more aware of how git LFS is hard to set up and use without conflicts.
